assets/js/hocs/with-transform-single-select-to-multiple-select.tsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore - Ignoring because @wordpress/element library does not have type definition for FunctionComponent
+// eslint-disable-next-line
+import { FunctionComponent } from '@wordpress/element';
+
+type SelectedOption = number | string | null | number[] | string[];
+
+interface OriginalComponentProps {
+	selected?: SelectedOption;
+}
+
+/**
+ * HOC that transforms a single select to a multiple select.
+ *
+ * @param {FunctionComponent< Record< string, unknown > >} OriginalComponent Component being wrapped.
+ */
+const withTransformSingleSelectToMultipleSelect = (
+	OriginalComponent: FunctionComponent< Record< string, unknown > >
+) => {
+	return ( props: OriginalComponentProps ): JSX.Element => {
+		let { selected } = props;
+		selected = selected === undefined ? null : selected;
+		const isNil = selected === null;
```

Comment:
When I looked at this again I see potential improvement, however, not mandatory, but just an idea.

`selected` is cast from `undefined` to `null` and eventually in line 33 to either empty array or `[ selected ]`.

What if that was simplified into a single step:

```suggestion
		selected = selected === undefined ? [] : [ selected ];
```

and then below:

```
			<OriginalComponent
				{ ...props }
				selected={ selectedArray }
			/>
```

I think it's easier to read, but if you disagree, feel free to keep `isNil`.

Reply:
Currently, I think there are 4 possible values for the `selected` prop:

1. A 'nil' value;
2. A single value;
3. An empty array;
4. An array with single/multiple values

My concern with the suggested approach is that if we have the third or the fourth possibilities, we will get as result a selected value like: `[ [] ]` or `[ [ 'value' ] ]`, which can break the application. 

I think that for now since we are just refactoring the files into Typescript we should avoid changing the current logic. I'm going to merge the current code, but if you think we should work on changing the logic for this component,  I'd be happy to work on this in a separate PR

Reply:
> My concern with the suggested approach is that if we have the third or the fourth possibilities, we will get as result a selected value like: [ [] ] or [ [ 'value' ] ], which can break the application.

I missed those cases! You're totally right!

> I think that for now since we are just refactoring the files into Typescript we should avoid changing the current logic. I'm going to merge the current code

Please merge, as I said, that was more of an idea, thanks for your response!